daisy/compute/compute.go

```diff
@@ -277,12 +277,23 @@ func (c *client) GetMachineType(project, zone, machineType string) (*compute.Mac
 }
 
 // ListMachineTypes gets a list of GCE MachineTypes.
-func (c *client) ListMachineTypes(project, zone string) (*compute.MachineTypeList, error) {
-	mt, err := c.raw.MachineTypes.List(project, zone).Do()
-	if shouldRetryWithWait(c.hc.Transport, err, 2) {
-		return c.raw.MachineTypes.List(project, zone).Do()
+func (c *client) ListMachineTypes(project, zone string) ([]*compute.MachineType, error) {
```

Comment:
I'm gonna be annoying and say "Can you refactor some of this logic into another method? There is a lot of copied code."

Maybe some method that does a generic paginated List that returns an untyped slice that you can then cast?

Reply:
I got this working but really hated it, I think being clever with reflect is unidiomatic:
https://blog.golang.org/laws-of-reflection

'It's a powerful tool that should be used with care and avoided unless strictly necessary.'